packages/amplify-graphql-function-transformer/src/graphql-function-transformer.ts

```diff
@@ -195,3 +255,256 @@ const lambdaArnKey = (name: string, region?: string, accountId?: string): string
+ */
+const validate = (config: FunctionDirectiveConfiguration, definition: FieldDefinitionNode, ctx: TransformerContextProvider): void => {
+  if (config.invocationType === 'Event') {
+    // For any occurance  of 'invocationType: Event' validate:
+    // is used on field where parent type is Mutation or Query.
+    validateIsSupportedEventInvocationParentType(config);
+
+    // The `@function` directive is defined as `repeatable` so chaining directives is valid.
+    // The following validations apply **only** if the final `@function` directive is defined
+    // with `invocationType: Event`.
+    //
+    // Valid
+    // type Mutation {
+    //   doStuff(msg: String): String
+    //   @function(name: 'foo', invocationType: Event)
+    //   @function(name: 'bar') // defaults to RequestResponse
+    // }
+    //
+    // Valid
+    // type Mutation {
+    //   doStuff(msg: String): EventInvocationResponse
+    //   @function(name: 'bar')
+    //   @function(name: 'foo', invocationType: Event)
+    // }
+    //
+    // Invalid -- response type must be `EventInvocationResponse`
+    // type Mutation {
+    //   doStuff(msg: String): String
+    //   @function(name: 'bar')
+    //   @function(name: 'foo', invocationType: Event)
+    // }
+    const ultimateFunctionDirectiveConfig = getUltimateFunctionDirectiveConfig(definition, ctx, config);
+    if (ultimateFunctionDirectiveConfig.invocationType === 'Event') {
+      // If `invocationType: Event` is defined in the final `@function` directive on a field:
+      // 2. return type of query / mutation is 'EventInvocationResponse'.
+      validateFieldResponseTypeForEventInvocation(definition, config);
+      // 3. shape of 'EventInvocationResponse' type defined in schema.
+      validateSchemaDefinedEventInvocationResponseShape(ctx.inputDocument);
+    }
+  }
+};
+
+const getUltimateFunctionDirectiveConfig = (
+  fieldDefinition: FieldDefinitionNode,
+  ctx: TransformerContextProvider,
+  currentDirectiveConfig: FunctionDirectiveConfiguration,
+): FunctionDirectiveConfiguration => {
+  const functionDirectivesForField = fieldDefinition.directives?.filter((directive) => directive.name.value === FunctionDirective.name);
+  if (!functionDirectivesForField || functionDirectivesForField.length === 0) {
+    // This really shouldn't happen because this code path should not execute without there being at least one `@function` directive
+    // defined on this field. But we're accessing the last element below, so better safe than sorry.
+    throw new InvalidDirectiveError(
+      'Unable to locate @function directives for field.' +
+        'This is an unexpected error that suggests there is a bug in the amplify-graphql-function-transformer.\n' +
+        'Please open a GitHub issue referencing this error message with your schema definition.',
+    );
+  }
+
+  const ultimateFunctionDirective = functionDirectivesForField[functionDirectivesForField.length - 1];
```

Comment:
I mentally read `ultimateFunctionDirective` in the "SUNDAY! SUNDAY! SUNDAY!" voice.